config/application.rb

```diff
@@ -201,6 +201,9 @@ def config.database_configuration
     $redis = DiscourseRedis.new
     Logster.store = Logster::RedisStore.new(DiscourseRedis.new)
 
+    require 'discourse_ip_info'
+    $ipinfo = DiscourseIpInfo.new
```

Comment:
I really really prefer to avoid globals here ... it can simply be a static hanging off DiscourseIpInfo... 

DiscourseIpInfo.lookup('10.0.0.1') ... then you cache it there. Also we need to make sure this does not explode if a maxmind db is missing, so have it do a file exists prior to loading it up.

Reply:
There is already [a check](https://github.com/discourse/discourse/pull/6335/files#diff-9c2f27b3cc391041329980be3a007b8fR8) if MaxMindDb is missing (it actually covers for the case when the file is corrupt too).